src/libostree/ostree-repo-pull.c

```diff
@@ -887,44 +875,25 @@ meta_fetch_on_complete (GObject           *object,
       goto out;
     }
 
-  if (fetch_data->is_detached_meta)
-    {
-      if (!ot_util_variant_map_fd (fd, 0, G_VARIANT_TYPE ("a{sv}"),
-                                   FALSE, &metadata, error))
-        goto out;
-
-      /* Now delete it, see comment in corresponding content fetch path */
-      (void) unlinkat (_ostree_fetcher_get_dfd (fetcher), temp_path, 0);
+  if (!ot_util_variant_map_fd (fd, 0, ostree_metadata_variant_type (objtype),
```

Comment:
Hm, it looks like we're unconditionally `mmap()ing` all objects?  That doesn't seem right.